lib/scanner.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const assert = require('assert');
+const objectScan = require('object-scan');
+
+// Scan object and pre-compute results for provided needles (see object-scan for syntax)
+// Results are accessed per needle through callback function 'onResult'
+
+module.exports = (haystack, needles) => {
+	const precomputed = needles.reduce((p, c) => Object.assign(p, { [c]: [] }), {});
```

Comment:
As this mutates `p` *anyways*, and as `precomputed` is local to the function, I prefer a for loop because of legibility.

Also, this uses `Object.create(null)` to allow for needles that clash with any field in the object prototype chain (e.g. `valueOf`).

```suggestion
	const precomputed = Object.create(null)
	for (const needle of needles) precomputed[needle] = []
```

Reply:
I'm ok with doing a loop here. I love to chain functions, which is why I usually don't use them.

Regarding `Object.create(null)`: I'm really not sure how this could possible cause any issues. Everything seems perfectly safe when using `{}` instead of Object.create(null). Can you please explain?

Reply:
If I understand correctly: If you forget to add a prototype needle you can still access this where the result is used without getting undefined. Is that what you meant?

Reply:
If you use a plain object with the default prototype chain, access using keys like `toString` & `valueOf` won't give `undefined`; `toString` is one of the more obvious cases where even authors of consuming code might be cautious, but there are more subtle cases.

To prevent this, one can either use a `Map`, or `Object.create(null)` to create an object without any prototype.